homeassistant/components/delijn/sensor.py

```diff
@@ -83,20 +83,23 @@ async def async_update(self):
         self._attributes["stopname"] = self._name
 
         try:
-            first = self.line.passages[0]
-            if first["due_at_realtime"] is not None:
-                first_passage = first["due_at_realtime"]
+            if len(self.line.passages) > 0: """do a check if there is a passage"""
```

Comment:
```suggestion
            if self.line.passages:
```

Reply:
Let move this line up out of the try... except block and invert the check and set availableto False and return early if true.

Reply:
Ok, I will replace the try and catch code by the length check this evening when returning to home :-)

Reply:
We still need the try except for the KeyError below but not for this line.

Reply:
Unless the library does a dict look up behind the scenes here that we don't see.

Reply:
I see in the library the following:
```
                                passages.append(
                                    {
                                        "passage": index,
                                        "line_number": linenumber,
                                        "direction": direction,
                                        "final_destination": final_destination,
                                        "due_at_schedule": due_at_schedule,
                                        "due_at_realtime": due_at_realtime,
                                        "due_in_min": due_in_min,
                                        "is_realtime": is_realtime,
                                        "line_number_public": linenumberpublic,
                                        "line_desc": linedesc,
                                        "line_transport_type": linetransporttype,
                                        "line_number_colourFront": linenumcolfront,
                                        "line_number_colourFrontHex": linenumcolfronthex,
                                        "line_number_colourBack": linenumcolback,
                                        "line_number_colourBackHex": linenumcolbackhex,
                                        "line_number_colourFrontBorder": linenumcolfrontborder,
                                        "line_number_colourFrontBorderHex": linenumcolfrontborderhex,
                                        "line_number_colourBackBorder": linenumcolbackborder,
                                        "line_number_colourBackBorderHex": linenumcolbackborderhex,
                                    }
```
So, is the try catch needed here?
In theory not, if the sync between HA core delijn and the pydelijn is not broken...
Practically, it does no harm to let the try catch in case keys are changed...
@Emilv2, what do you think?

Meanwhile I've tested locally the last code change (= invert the if and set it before the try-catch + adding the Keyerror in the logfile): no errors.

Reply:
There's no risk with access the `line.passages` property since it just returns the attribute that holds the list of passages. So we don't need to wrap that in the try... except. :+1: